money_honey.py

```diff
@@ -1,80 +1,164 @@
+                    else:
+                        print("You failed.")
+                        self.handle_input_currency()
+                # if he doesn't care me neither
+                else:
+                    self.args.input_currency = currency_codes[randint(0, (len(currency_codes) - 1))]
+            # if symbol has just one currency don't bother user, just get rid of array
+            else:
+                self.args.input_currency = self.get_first_value(currency_codes)
+        # if it's not symbol just make it upper so it looks nice in output
         else:
-            print(output_string_start)
-            for cur, ca in converted_amounts.items():
-                print("\t" + output_string_end(ca, cur))
-
-
-def get_first_value(my_list):
-    return next(iter(my_list))
-
-
-def output_string_end(converted_amount, output_currency):
-    return str(converted_amount) + " " + output_currency
+            self.args.input_currency = self.args.input_currency.upper()
 
+    def handle_output_currency(self):
+        # if currency is symbol get currencies to which to convert
+        if len(self.args.output_currency) != 3:
+            self.args.output_currency = self.convert_symbol(self.args.output_currency)
+        # if it's char code just make sure it's upper and in array
+        else:
+            self.args.output_currency = [self.args.output_currency.upper()]
+
+    def convert_symbol(self, symbol):
+        """
+        Converts symbol to currency char codes. Shouldn't be used before arguments_parser().
+        :param symbol: Symbol we want to convert.
+        :return: Array with currency codes for symbol.
+        """
+        return self.currency_changer.currency_signs[symbol]
+
+    def print_results(self):
+        if self.args.no_json:
+            output_string_start = "\n" + str(self.args.amount) + " " + self.args.input_currency + " is"
+            if len(self.converted_amounts) == 1:
+                print(output_string_start + " " +
+                      self.output_string_end(format(self.get_first_value(self.converted_amounts.values()), '.4f'),
+                                             self.get_first_value(self.converted_amounts.keys())) +
+                      ".")
+            else:
+                print(output_string_start)
+                for currency, currency_amount in self.converted_amounts.items():
+                    print("\t" + self.output_string_end(format(currency_amount, '>20.4f'), currency))
+        else:
+            output_json = {}
 
-def check_currency(name, currency):
-    if CurrenciesChanger.is_valid_currency(currency):
-        return currency
-    else:
-        print("Couldn't parse " + name[2:] + ". Either it isn't supported or you messed up on your part.")
-        sys.exit(1)
+            converted_currencies = {}
+            for currency, currency_amount in self.converted_amounts.items():
+                converted_currencies[currency] = format(currency_amount, '.4f')
```

Comment:
I am always formatting just single value and they are spread over string building mess so I felt like this is the way.